Automattic/jetpack
# JS Components: Convert Button component to TypeScript

#### Changes proposed in this Pull Request:

* This PR converts `Button` component to TypeScript and cleans up the propTypes.
* It also converts the button stories to TS

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the stories of `Button` component work as expected
* Check the consumers of `Button` and related components and confirm that they work as expected
* Check for any regression
